app/assets/stylesheets/govuk_publishing_components/components/_cookie-banner.scss

```diff
@@ -8,18 +8,16 @@ $govuk-cookie-banner-background: govuk-colour("light-grey", "grey-4");
 }
 
 .gem-c-cookie-banner {
-  @include govuk-font($size: 16);
-  padding: govuk-spacing(2) 0;
   background-color: $govuk-cookie-banner-background;
 }
 
-// Only show accept button if users have js and can accept
-.gem-c-cookie-banner__button-accept {
+// Only show buttons if users have js and can accept
+.gem-c-button.js-visible {
```

Comment:
Can you explain the logic here a bit? I'm confused why we have JS visible rules for individual elements of the cookie banner, when the cookie banner itself isn't shown when JS is disabled.

Reply:
The banner is shown when JS is disabled, but the buttons aren't. I don't know why that decision was made but it's the way it's worked for a long time. It links to the cookies page which tells you you need js to set your preferences. 

So the flow is: show the banner by default. If js is on, hide it with styling based on the js-enabled class, while the more specific js re-displays it if necessary based on appropriate cookie reading logic.

Reply:
That's really interesting, thanks. I think it would be worth documenting this behaviour somewhere, perhaps in the component guide page?

Testing this in the preview app the whole component disappears if JS is disabled. Is that expected?

Reply:
No... I've got it running on local static/government-frontend and that doesn't happen. :(

Reply:
Whatever's going wrong is also happening on the live version, needs further investigation, but won't block this.